core/shared/src/main/scala/laika/parse/code/languages/YAMLSyntax.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2012-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package laika.parse.code.languages
+
+import cats.data.NonEmptyList
+import laika.ast.CodeSpan
+import laika.bundle.SyntaxHighlighter
+import laika.parse.builders._
+import laika.parse.code.common.{Comment, Identifier, NumberLiteral, StringLiteral}
+import laika.parse.code.{CodeCategory, CodeSpanParser}
+import laika.parse.implicits._
+import laika.parse.text.TextParsers.{literal, oneOf}
+
+/**
+ * @author Michał Sitko
+ */
+object YAMLSyntax extends SyntaxHighlighter {
+  /** The names of the language (and its optional aliases) as used in text markup */
+  override def language: NonEmptyList[String] = NonEmptyList.of("yaml", "yml")
+
+  private val keyName = Identifier.alphaNum.withIdStartChars('_', '-', '.').withCategory(CodeCategory.AttributeName)
+
+  private val valueLiteral = {
+    val doubleQuotedStrLiteral = StringLiteral.singleLine('"')
+    val singleQuotedStrLiteral = StringLiteral.singleLine('\'')
+    val number = NumberLiteral.decimalFloat | NumberLiteral.decimalInt
+    val boolean = (literal("true") | literal("false")).map(b => Seq(CodeSpan(b, CodeCategory.BooleanLiteral)))
+    val nullLiteral = (literal("null")).map(n => Seq(CodeSpan(n, CodeCategory.LiteralValue)))
+    val unquotedStrLiteral = delimitedBy(literal(" #") | literal("\n")).acceptEOF.keepDelimiter.map(s => List(CodeSpan(s, CodeCategory.StringLiteral)))
+
+    doubleQuotedStrLiteral | singleQuotedStrLiteral | number | boolean | nullLiteral | unquotedStrLiteral
+  }
+
+  private val keyAndValue = {
+    val attributeName = {
+      // https://docs.ansible.com/ansible/latest/reference_appendices/YAMLSyntax.html#yaml-basics:
+      // the colon must be followed by a space
+      (keyName ~ oneOf(':') ~ oneOf(' ','\t')).map { r =>
+        List(r._1._1, CodeSpan(r._1._2 + r._2))
+      }
+    }
```

Comment:
The actual syntax for key+value is much less restrictive. The precise grammar is zero or more spaces before the colon and one of more after it.

I'd suggest the following alternative (untested, but probably fine):
```scala
private val keyAndValue = {
  val separator = (anyOf(' ','\t') ~ ":" ~ someOf(' ','\t')).source.map(s => List(CodeSpan(s)))
  (keyName ~ separator ~ valueLiteral).concat
}
```

Reply:
Thanks, it worked perfectly